iree/compiler/Conversion/LinalgToSPIRV/Utils.cpp

```diff
@@ -79,9 +89,14 @@ Optional<Value> allocateWorkgroupMemory(OpBuilder &b, SubViewOp subview,
 }
 
 LogicalResult deallocateWorkgroupMemory(OpBuilder &b, Value buffer) {
-  auto allocOp = buffer.getDefiningOp<AllocOp>();
-  b.create<DeallocOp>(allocOp.getLoc(), buffer);
-  return success();
+  // There is no utility of an explicit deallocation (as of now). Instead the
+  // workgroup memory is effectively stack memory that is automatically dead at
+  // the end of the function. The SPIR-V lowering treats such deallocs as
+  // no-ops. So dont insert it in the first place, rather just check that the
+  // deallocation is for workgroup memory.
+  MemRefType bufferType = buffer.getType().dyn_cast<MemRefType>();
+  if (!bufferType) return failure();
```

Comment:
nit: is this case possible? Should it be an assert?

Reply:
Its a callback so not sure I want to assert. This returns failure which should be handled by the caller.